.github/workflows/build.yml

```diff
@@ -134,38 +105,37 @@ jobs:
           message("::set-output name=install_prefix::${PREFIX}")
           message("::set-output name=install_dir::${INSTALL_DIR}")
 
-      - name: 'Build with CMake and Ninja'
+      - name: 'Windows: set up developer environment'
+        uses: ilammy/msvc-dev-cmd@v1
+        if: matrix.config.os == 'windows-latest'
+
+      - name: 'Configure with CMake and vcpkg.'
+        run: cmake --preset ${{ matrix.config.cmake_preset }} -DCMAKE_INSTALL_PREFIX="${{ steps.cmake_install_prefix.outputs.install_prefix }}"
         env:
-          CCACHE_BASEDIR: '${{ github.workspace }}'
-          CCACHE_DIR: '${{ runner.workspace }}/.ccache'
-          CCACHE_COMPRESS: 'true'
-          CCACHE_COMPRESSLEVEL: '6'
-          CCACHE_MAXSIZE: '500M'
-        uses: lukka/run-cmake@v3
-        with:
-          cmakeListsOrSettingsJson: CMakeListsTxtAdvanced
-          cmakeListsTxtPath: '${{ github.workspace }}/CMakeLists.txt'
-          useVcpkgToolchainFile: ${{ matrix.config.os != 'ubuntu*' }}
-          buildDirectory: '${{ runner.workspace }}/b/ninja'
-          cmakeAppendedArgs: '-DCMAKE_INSTALL_PREFIX="${{ steps.cmake_install_prefix.outputs.install_prefix }}" ${{ matrix.config.cmake_config }}'
+          VCPKG_BINARY_SOURCES: 'clear;files,${{ env.VCPKG_CACHE_LOCATION }},readwrite'
```

Comment:
the syntax of this is quite non-obvious; maybe worth a comment?

Reply:
Good point, will add a link to the vcpkg docs that explain it as I missed it several times myself.